internal/network/portgroup_subresource.go

```diff
@@ -0,0 +1,70 @@
+/*
+ *  Copyright 2023 VMware, Inc.
+ *    SPDX-License-Identifier: MPL-2.0
+ */
+
+package network
+
+import (
+	"fmt"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/validation"
+	validation_utils "github.com/vmware/terraform-provider-vcf/internal/validation"
+	"github.com/vmware/vcf-sdk-go/models"
+	"strings"
+)
+
+// PortgroupSchema this helper function extracts the Portgroup Schema, so that
+// it's made available for both Domain and Cluster creation.
+func PortgroupSchema() *schema.Resource {
+	return &schema.Resource{
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Description:  "Port group name",
+				ValidateFunc: validation.NoZeroValues,
+			},
+			"transport_type": {
+				Type:     schema.TypeString,
+				Required: true,
+				Description: "Port group transport type, One among: VSAN, VMOTION, MANAGEMENT, PUBLIC, " +
+					"NFS, VREALIZE, ISCSI, EDGE_INFRA_OVERLAY_UPLINK",
+				ValidateFunc: validation.StringInSlice([]string{
```

Comment:
Should this be a var and then:

```go
ValidateFunc: validation.StringInSlice(PortGroupTransportTypeAllowedValues,true),
```

Reply:
vars can be changed, []string cannot be a constant